src/AggregateFunctions/AggregateFunctionQuantileGK.cpp

```diff
@@ -268,11 +268,7 @@ class ApproxSampler
 
         size_t sampled_len = 0;
         readBinaryLittleEndian(sampled_len, buf);
-        if (sampled_len > compress_threshold)
-            throw Exception(
-                ErrorCodes::INCORRECT_DATA, "The number of elements {} for quantileGK exceeds {}", sampled_len, compress_threshold);
-
-        sampled.resize(sampled_len);
+        sampled.resize_exact(sampled_len);
```

Comment:
A malformed state will make you request a huge amount of memory. How can it happen normally?

Reply:
I could reproduce it easily in https://github.com/oap-project/gluten/pull/4829 if checking in `quantileGK` exists. But currently I can't reproduce in clickhouse-server...

Reply:
> A malformed state will make you request a huge amount of memory. How can it happen normally?


To be precise, it is not a huge amount of memory (11869). It seems not related with overflow. 
```
0: jdbc:hive2://localhost:10000/> select l_linenumber % 10, approx_percentile(l_extendedprice, 0.5) from lineitem group by l_linenumber % 10;  
Error: org.apache.hive.service.cli.HiveSQLException: Error running query: org.apache.spark.SparkException: Job aborted due to stage failure: Task 11 in stage 2.0 failed 1 times, most recent failure: Lost task 11.0 in stage 2.0 (TID 6) (bigo executor driver): io.glutenproject.exception.GlutenException: io.glutenproject.exception.GlutenException: 
The number of elements 11869 for quantileGK exceeds 10000
0. ./contrib/llvm-project/libcxx/include/exception:141: Poco::Exception::Exception(String const&, int) @ 0x0000000012067b9d in /data1/liyang/cppproject/kyli/ClickHouse/build_gcc/utils/extern-local-engine/libch.so
1. ./build_gcc/./src/Common/Exception.cpp:96: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000b03b0bf in /data1/liyang/cppproject/kyli/ClickHouse/build_gcc/utils/extern-local-engine/libch.so
2. ./contrib/llvm-project/libcxx/include/string:1499: DB::Exception::Exception<unsigned long&, unsigned long&>(int, FormatStringHelperImpl<std::type_identity<unsigned long&>::type, std::type_identity<unsigned long&>::type>, unsigned long&, unsigned long&) @ 0x0000000006ccf93e in /data1/liyang/cppproject/kyli/ClickHouse/build_gcc/utils/extern-local-engine/libch.so
3. ./build_gcc/./src/AggregateFunctions/AggregateFunctionQuantileGK.cpp:0: DB::(anonymous namespace)::QuantileGK<double>::deserialize(DB::ReadBuffer&) @ 0x000000000bfb0fac in /data1/liyang/cppproject/kyli/ClickHouse/build_gcc/utils/extern-local-engine/libch.so
```

Reply:
> We can't make sure that size of sampled doesn't exceed default_compress_threshold, after two states are merged together and then compressed

Shouldn't merge or compress discard the extra samples? Otherwise the state will keep growing and growing indefinitely. What's the point of having a threshold if we are not going to respect it?

Reply:
I'm not familiar with the internals of the algorithm...  Let me dig into it for a while

Reply:
Yes, spark added it too. I think compress_threshold is an extra option introduced in engineering to make samples to be compressed while inserting continuously.

Reply:
The question is why was it added? Was it knowing the reason or just because we copied Spark code?

Anyway, it seems we can remove the check

Reply:
it was from spark implementation